include/merlin_hashtable.cuh

```diff
@@ -148,14 +149,15 @@ static constexpr auto& thrust_par = thrust::cuda::par;
  *
  */
 template <typename K, typename V, typename S = uint64_t,
-          typename ArchTag = Sm80>
+          typename ArchTag = Sm80, typename Allocator = DefaultAllocator>
```

Comment:
In some case like tensorflow, the allocator is created before merlin::HashTable construction, which created in OpKernel running. So It would be better to allow assignment from outside allocator, whom match the interface of internal.

Reply:
Sure, I though of this, please move to this one: https://github.com/NVIDIA-Merlin/HierarchicalKV/pull/145